crates/core/src/commands/check.rs

```diff
@@ -37,7 +38,38 @@ pub struct CheckOptions {
     pub read_data: bool,
 }
 
-impl CheckOptions {
+#[derive(Debug)]
+pub struct CheckResults {
+    pub errors: Vec<(LogLevel, CheckError)>,
```

Comment:
Ah, I see now, this is the reason it is part of the public API, I guess. Hmm, maybe we can wrap it in our own LogLevel type or something, so we could essentially just swap log out against tracing for example under the hood?